keymapper/injection/consumers/consumer.py

```diff
@@ -52,10 +52,10 @@ def is_enabled(self):
         """Check if the consumer will have work to do."""
         raise NotImplementedError
 
-    def write(self, key):
+    def write(self, key, target="keyboard"):
         """Shorthand to write stuff."""
-        self.context.uinput.write(*key)
-        self.context.uinput.syn()
+        self.context.uinput_devices[target].uinput.write(*key)
+        self.context.uinput_devices[target].uinput.syn()
 
     def forward(self, key):
```

Comment:
I just tried doing the combination modifier fix that involves adding a `k(Shift_L)`, but with `e(EV_KEY, KEY_LEFTSHIFT, 0)`.

Using `e` for that is only possible if the forward function also uses the `uinput_devices["keyboard"].uinput` device